docs/middleware/op-submission.md

```diff
@@ -61,6 +67,10 @@ backend.use('apply', (context, next) => {
   if (userId !== ownerId) {
     return next(new Error('Unauthorized'))
   }
+
+  // Add op metadata to snapshot before snapshot is stored
+  Object.assign(context.snapshot.m, context.op.m);
```

Comment:
Okay have discussed with @ericyhwang and we think:

- we should go back to having a feature flag (sorry about the U-turn!)
- if the feature flag is enabled, we send all metadata across pubsub
- if the feature flag is enabled, we also apply the metadata projection in the `op` middleware, which should handle the cases I asked for on your tests (reiterated here for clarity)

Cases we need to handle:
- Op submission (the case you're already testing): ops broadcast over pubsub to other subscribers
- Op fetching: have a stale document that calls `doc.fetch()` — this will fetch ops through [`backend._getSanitizedOps()`](https://github.com/share/sharedb/blob/cc0e3382bf3df4c38c7ccc6bc52da71aff8c9f74/lib/agent.js#L608-L610)
- Ops sent back when submitting: have a stale document that calls `doc.submitOp()` — this retrieves ops through another mechanism in [`SubmitRequest.submit()`](https://github.com/share/sharedb/blob/cc0e3382bf3df4c38c7ccc6bc52da71aff8c9f74/lib/submit-request.js#L106)